libraries/Microsoft.Bot.Builder.LanguageGeneration/Evaluator.cs

```diff
@@ -331,7 +331,7 @@ private bool EvalExpressionInCondition(string exp)
             }
         }
 
-        private object EvalExpression(string exp)
+        private object EvalExpression(string exp, bool tolerant = true)
```

Comment:
It's good to have option here, so that we can always revisit, or choose different policy in different context. 

maybe we can call this `tolerantNull`, might be more specific.  or 'nullable'. "tolerant" is a a little bit too general